controllers/ie_controller.go

```diff
@@ -0,0 +1,81 @@
+package controllers
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+
+	mgo "gopkg.in/mgo.v2"
+
+	"github.com/gin-gonic/gin"
+)
+
+// IEController Wrapped for Patient Collection
+type IEController struct {
+	DB             *mgo.Database
+	CollectionName string
+	Item           interface{}
```

Comment:
We don't have to use the generic controller for everything. But for the many cases where were doing simple crud actions, I think it makes sense. My issue with moving away from this is that I don't want to have repeated code everywhere for these controllers that are substantially the same. If we can accomplish DRY code with individual controllers, then I'm happy to do that.